### DISCUSSION TITLE
How can I override parser options for each collection type?

### DISCUSSION THREAD

**Discussion by ESnark** [strapi/strapi#6306] [Legacy Topics]
I'm trying to differentiate `maxFileSize` config of a media type field from `request.json` and I couldn't find out the way or a possibility.

**Comment by derrickmehaffy** [comment]
If you are on the beta, you can see the example in my project here: https://github.com/canonn-science/CAPIv2-Strapi/blob/0ac44cf612d714db2a4fd2afd151cd71f73de5e8/config/environments/development/request.json#L19-L24

Likewise if you are on the rc.x releases you can see my WIP example here: https://github.com/canonn-science/CAPIv2-Strapi/blob/5e6997d7b9ee6e434042b8b6448022eb6d80138e/config/middleware.js#L55-L60

**Reply by derrickmehaffy** [reply]
maxFileSize has no bearing on the parser because the files aren't being "parsed"

**Reply by ESnark** [reply]
First, I'm on the 3.0.0-beta.20.3 and actually my maxFileSize config is working pretty well.
Here is my `~/config/environment/**/request.json`
```json
{
  "session": {  },
  "loger": {  },
  "parser": {
    "enabled": true,
    "multipart": true,
    "formidable": {
      "maxFileSize": 10485760
    }
  }
}
```
But aside from this config thing, I realized that not only was I completely misunderstanding the concept of strapi's media library, it was pointless to limit the file size to upload on each field.

Maybe I can solve the problem by focusing on the process of selecting files for media field, not the aspect of upload.
